test/StripeResource.spec.js

```diff
@@ -32,4 +35,245 @@ describe('StripeResource', function() {
       expect(headers).to.not.include.keys('Stripe-Version');
     });
   });
+
+  describe('Retry Network Requests', function() {
+    // Use a real instance of stripe as we're mocking the http.request responses.
+    var realStripe = require('../lib/stripe')(utils.getUserStripeKey());
+    var options = {
+      host: stripe.getConstant('DEFAULT_HOST'),
+      path: '/v1/charges',
+      data: {
+        amount: 1000,
+        currency: 'usd',
+        source: 'tok_visa',
+        description: 'test'
+      },
+      params: 'amount=1000&currency=usd&source=tok_visa&description=test'
+    };
+
+    afterEach(function() {
+      realStripe.setMaxNetworkRetries(0);
+      stripe.setMaxNetworkRetries(0);
+    });
+
+    after(function() {
+      nock.cleanAll();
+    })
+
+    describe('_request', function() {
+      it('throws an error on connection failure', function(done) {
```

Comment:
is this a connection error, or an api error?

Reply:
Connection error, which is simulated via nock's `replyWithError` method.